baybe/campaign.py

```diff
@@ -253,14 +279,25 @@ def recommend(self, batch_quantity: int = 5) -> pd.DataFrame:
         return rec
 
 
-def _unstructure_with_version(obj: Any) -> dict:
+def _add_version(dict_: dict) -> dict:
     """Add the package version to the created dictionary."""
     from baybe import __version__
 
-    return {
-        **converter.unstructure_attrs_asdict(obj),
-        "version": __version__,
-    }
+    return {**dict_, "version": __version__}
 
 
-converter.register_unstructure_hook(Campaign, _unstructure_with_version)
+# Register de-/serialization hooks
+unstructure_hook = cattrs.gen.make_dict_unstructure_fn(
```

Comment:
Why do we need to have new hooks here and what do they do?

Reply:
I guess it makes sure the deprecated (and hence still existing) flag is not de/serialzied + adds the baybe version to the json (not new in this PR though)

Reply:
That makes sense, can you confirm @AdrianSosic ?

Reply:
Yes and no. "Yes" because that one line, i.e.
```python
numerical_measurements_must_be_within_tolerance=cattrs.override(omit=True)
```
is to avoid adding the deprecated attribute to the unstructured dict. 

"No", because the the above is not even strictly necessary (but certainly better to have). However, the actual reason for the new hooks is to instruct cattrs how to handle the new private unexposed attributes. Because they are not exposed, they can no longer be structured by going the standard route via the constructor. However, cattrs takes care of it internally if you tell it to, which is what I did here.